### ISSUE TITLE
Cannot activate any other plugin

### ISSUE THREAD

**Issue by adrianofnatal** [WebDevStudios/NextGen-Public-Uploader#49]
Before I activate the NextGen Public Uploadar, I can't activate any other plugin without deactivate it.
This is the error I got

[Fri Nov 20 09:54:21 2015] [error] [client 192.168.1.93] PHP Warning:  include_once(NGGALLERY_ABSPATHlib/ngg-db.php) [<a href='function.include-once'>function.include-once</a>]: failed to open stream: No such file or directory in /var/www/somesite/wp-content/plugins/NextGen-Public-Uploader/nextgen-public-uploader.php on line 245, referer: http://192.168.1.240/somesite/wp-admin/plugins.php?plugin_status=all&paged=1&s
[Fri Nov 20 09:54:21 2015] [error] [client 192.168.1.93] PHP Warning:  include_once() [<a href='function.include'>function.include</a>]: Failed opening 'NGGALLERY_ABSPATHlib/ngg-db.php' for inclusion (include_path='.:/usr/share/pear:/usr/share/php') in /var/www/somesite/wp-content/plugins/NextGen-Public-Uploader/nextgen-public-uploader.php on line 245, referer: http://192.168.1.240/somesite/wp-admin/plugins.php?plugin_status=all&paged=1&s
[Fri Nov 20 09:54:21 2015] [error] [client 192.168.1.93] PHP Fatal error:  Class 'nggdb' not found in /var/www/somesite/wp-content/plugins/NextGen-Public-Uploader/nextgen-public-uploader.php on line 247, referer: http://192.168.1.240/somesite/wp-admin/plugins.php?plugin_status=all&paged=1&s

**Comment by adrianofnatal** [comment]
Sorry, I mean "after" I activate NG Public Uploader.

**Comment by tw2113** [comment]
What version of NextGEN Gallery are you using? Do note that our plugin is not updated for NGG 2.0.x still, and is still coded for the legacy 1.9.x line. It's possible they've started to break very old compatibility at the moment.

**Comment by tw2113** [comment]
Also, just in case, is NextGEN Gallery itself active first?

**Comment by adrianofnatal** [comment]
Hi.
Im using version 2.1.15 of NG Gallery.
Yes, NG Gallery is active first. The integration with NG Gallery and NG
Public Uploader is ok, I can send photos, approve, and build a page with
this gallery. The problem is when the NG Public Uploader is active, I can't
activate any other plugins. I need to deactivate NG Public Uploader first,
activate the other plugin and activate the NG Public.

Thanks!

2015-11-20 18:41 GMT-02:00 Michael Beckwith <notifications@github.com>:

> Also, just in case, is NextGEN Gallery itself active first?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebDevStudios/NextGen-Public-Uploader/issues/49#issuecomment-158518585>
> .
>



-- 
Adriano Felippeto Natal

**Comment by tw2113** [comment]
Did you download a copy of the plugin from here on GitHub within the past 4 days? That's the only way I can see that your plugin folder would have capitalization matching the repo here, and the master branch copy has code that would line up with the errors you're seeing above.

The "stable" release that we have on wordpress.org doesn't and line 245 there is an H2 tag.

I'm going to have to say that the master branch here is not "stable" yet, especially after our PR from earlier this week that needs some testing/confirmation of things working still.

**Comment by adrianofnatal** [comment]
I have downloaded yesterday.


2015-11-21 17:32 GMT-02:00 Michael Beckwith <notifications@github.com>:

> Did you download a copy of the plugin from here on GitHub within the past
> 4 days? That's the only way I can see that your plugin folder would have
> capitalization matching the repo here, and the master branch copy has code
> that would line up with the errors you're seeing above.
>
> The "stable" release that we have on wordpress.org doesn't and line 245
> there is an H2 tag.
>
> I'm going to have to say that the master branch here is not "stable" yet,
> especially after our PR from earlier this week that needs some
> testing/confirmation of things working still.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebDevStudios/NextGen-Public-Uploader/issues/49#issuecomment-158676171>
> .
>



-- 
Adriano Felippeto Natal

**Comment by tw2113** [comment]
Can you see if you can get everything working as necessary with the version from WordPress.org, while we do what is necessary with our copy of the plugin here on GitHub?

**Comment by adrianofnatal** [comment]
I have installed te version from WordPress.org plugin first. The approval
of uploaded pictures was not working.
After this I have found the Github version, I installed and approval of
uploaded images is working.
The only problem is that I can't activate any other plugin with NG Public
Uploader active.
I deactivate it before I activate other plugins then I activated NG Public
Uploader back.

2015-11-21 19:57 GMT-02:00 Michael Beckwith <notifications@github.com>:

> Can you see if you can get everything working as necessary with the
> version from WordPress.org, while we do what is necessary with our copy of
> the plugin here on GitHub?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebDevStudios/NextGen-Public-Uploader/issues/49#issuecomment-158684971>
> .
>



-- 
Adriano Felippeto Natal

**Comment by tw2113** [comment]
Yes, I have not forgotten the original issue that you can't activate other plugins afterwards. That part isn't making sense to me though, and I've been making sure if you couldn't be using the proper release at the moment.

I don't know why, but it isn't falling on my dev install and I can activate others just fine.

The only issue I'm potentially seeing in your posted errors is a missing forward slash in the path, between the NGG_ABSPATH and the rest of the file name to include. Are you willing to try adding that yourself? It'd be on that line 245 like listed in the errors

**Comment by adrianofnatal** [comment]
I could try, for sure.
I'll tell you later.

Thanks.

2015-11-22 16:31 GMT-02:00 Michael Beckwith <notifications@github.com>:

> Yes, I have not forgotten the original issue that you can't activate other
> plugins afterwards. That part isn't making sense to me though, and I've
> been making sure if you couldn't be using the proper release at the moment.
>
> I don't know why, but it isn't falling on my dev install and I can
> activate others just fine.
>
> The only issue I'm potentially seeing in your posted errors is a missing
> forward slash in the path, between the NGG_ABSPATH and the rest of the file
> name to include. Are you willing to try adding that yourself? It'd be on
> that line 245 like listed in the errors
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebDevStudios/NextGen-Public-Uploader/issues/49#issuecomment-158787406>
> .
>



-- 
Adriano Felippeto Natal

**Comment by adrianofnatal** [comment]
Hi. Sorry about the delay.
Your previous solution does not worked.

2015-11-22 18:03 GMT-02:00 Adriano Natal <adrianofnatal@gmail.com>:

> I could try, for sure.
> I'll tell you later.
>
> Thanks.
>
> 2015-11-22 16:31 GMT-02:00 Michael Beckwith <notifications@github.com>:
>
>> Yes, I have not forgotten the original issue that you can't activate
>> other plugins afterwards. That part isn't making sense to me though, and
>> I've been making sure if you couldn't be using the proper release at the
>> moment.
>>
>> I don't know why, but it isn't falling on my dev install and I can
>> activate others just fine.
>>
>> The only issue I'm potentially seeing in your posted errors is a missing
>> forward slash in the path, between the NGG_ABSPATH and the rest of the file
>> name to include. Are you willing to try adding that yourself? It'd be on
>> that line 245 like listed in the errors
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/WebDevStudios/NextGen-Public-Uploader/issues/49#issuecomment-158787406>
>> .
>>
>
>
>
> --
> Adriano Felippeto Natal
>



-- 
Adriano Felippeto Natal

**Comment by tw2113** [comment]
Has anything changed between now and last November? I have to believe there were some plugin updates to NGG core since.

Are you on windows or linux hosting?

**Comment by adrianofnatal** [comment]
Im using linux hosting. Always used.

I have download lastest version of NGG Public Uploader from github, and
problem persist.
Hmm, in my server nothing changes. I'm using Wordpress 4.4.2, but in other
versions the problem occurs.

Thanks!


2016-02-17 17:30 GMT-02:00 Michael Beckwith <notifications@github.com>:

> Has anything changed between now and last November? I have to believe
> there were some plugin updates to NGG core since.
>
> Are you on windows or linux hosting?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebDevStudios/NextGen-Public-Uploader/issues/49#issuecomment-185367100>
> .
>



-- 
Adriano Felippeto Natal

**Comment by tw2113** [comment]
At this point, I honestly have no idea why.

**Comment by adrianofnatal** [comment]
The error does not happen in your environment?
You need to activete the NGG Photo Uploader, then activate some other
plugin. When you activate other plugin the error is shown.



2016-02-23 2:51 GMT-03:00 Michael Beckwith <notifications@github.com>:

> At this point, I honestly have no idea why.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebDevStudios/NextGen-Public-Uploader/issues/49#issuecomment-187550316>
> .
>



-- 
Adriano Felippeto Natal

**Comment by tw2113** [comment]
Nope, I'm still able to activate other things fine without any errors appearing.

**Comment by adrianofnatal** [comment]
I used the same approach wich NGGallery uses, for loading its plugin.

I add the following methods in nextgen-public-uploader.php, inside class
NGGallery_Public_uploader:

   function is_not_activating()
    {
        return !$this->is_activating();
    }

    function is_activating()
    {
        $retval =  strpos($_SERVER['REQUEST_URI'], 'plugins.php') !== FALSE
&& isset($_REQUEST['action']) && $_REQUEST['action'] == 'activate';
        if (!$retval && strpos($_SERVER['REQUEST_URI'], 'update.php') !==
FALSE && isset($_REQUEST['action']) && $_REQUEST['action'] ==
'install-plugin' && isset($_REQUEST['plugin']) &&
strpos($_REQUEST['plugin'], 'nextgen-public-uploader') === 0) {
            $retval = TRUE;
        }
        if (!$retval && strpos($_SERVER['REQUEST_URI'], 'update.php') !==
FALSE && isset($_REQUEST['action']) && $_REQUEST['action'] ==
'activate-plugin' && isset($_REQUEST['plugin']) &&
strpos($_REQUEST['plugin'], 'nextgen-public-uploader') === 0) {
            $retval = TRUE;
        }

        return $retval;
    }

Then I modify the constructor, adding the lines between
register_activation_hook(... and the last add_action(... inside the "if"
below:

if ($this->is_not_activating()) {
....
}

This way, I ensure that code won't run when was activating other plugins
than Nextgen Public Uploader plugin.

The error is gone.

2016-02-23 12:31 GMT-03:00 Michael Beckwith <notifications@github.com>:

> Nope, I'm still able to activate other things fine without any errors
> appearing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebDevStudios/NextGen-Public-Uploader/issues/49#issuecomment-187744995>
> .
>



-- 
Adriano Felippeto Natal

**Comment by tw2113** [comment]
At this point whatever works for you is encouraged. :)

Still don't know why it was being run so early for you and not for myself.
